proxylib/npds/client_test.go

```diff
@@ -27,8 +28,8 @@ func Test(t *testing.T) {
 }
 
 type ClientSuite struct {
-	acks  int
-	nacks int
+	acks  uint64
```

Comment:
I think you are having to make this an atomic variable just to synchronize the goroutine write and reads in the assert call in the test case. Again, nyc, but it looks like the the test case should've synchronized the calls via a channel instead of adding sleep. The callback function in `UpsertNetworkPolicy` can signal on a channel that the test case blocks on. Anyway I can revisit this in a separate PR. 

I don't really understand the sleep duration specified in this block - 

```
// Create version 1 with resource 0.
	s.UpsertNetworkPolicy(c, xdsServer, resources[0])

	time.Sleep(DialDelay * BackOffLimit) ------> there is no back-off?
	c.Assert(s.acks, Equals, 1)
	c.Assert(s.nacks, Equals, 0)
```

Reply:
Fully agreed, I have the same feeling here. But I haven't spend the time to understand it yet. 

I'll leave this open for now (as there is no urgency to merge it). If I find time in the next few to clean this up, I will do it in this PR, otherwise I'll label it ready-to-merge (assuming this doesn't need a Jenkins run, since it's unit tests only).